### PR TITLE
Add async support with NeoVim

### PR DESCRIPTION
**Pull request by idanarye**

### CODE REVIEW

File: plugin/ctrlsf.vim [hunk 1]
```diff
@@ -230,7 +230,7 @@ endif
 
 " g:ctrlsf_search_mode {{{2
 if !exists('g:ctrlsf_search_mode')
-    if v:version < 800 || (v:version == 800 && !has('patch1039')) || has('nvim')
+    if v:version < 800 || (v:version == 800 && !has('patch1039') && !has('nvim'))
```
[the remark this answers is not in the document]
Comment (idanarye) [reply plugin/ctrlsf.vim]: Not sure if NeoVim _has_ a version of less than 800, but it did have async from day one so yea, checking for version is redundant.

File: autoload/ctrlsf/async.vim [hunk 2]
```diff
@@ -83,14 +83,22 @@ endf
...
-                \ 'out_mode': 'nl',
-                \ 'err_mode': 'raw',
-                \ 'in_io': 'null',
-                \ })
+    if has('nvim')
+        let s:job_id = jobstart(a:command, {
+                    \ 'on_stdout': "ctrlsf#async#NeoVimSearchCBWrapper",
+                    \ 'on_stderr': "ctrlsf#async#NeoVimSearchCBWrapper",
+                    \ 'on_exit': "ctrlsf#async#NeoVimSearchCBWrapper",
+                    \ })
```
[the remark this answers is not in the document]
Comment (idanarye) [reply autoload/ctrlsf/async.vim]: Tested it now - and it works. NeoVim does not have anything equivalent to `in_io`.

File: autoload/ctrlsf/buf.vim [hunk 3]
```diff
@@ -40,7 +40,28 @@ endf
...
+        " Unlike setbufline, nvim_buf_set_lines can only accept a list
+        if type(a:content) == type('')
+            let l:content = split(a:content, '\v(\r\n)|\n')
+        else
+            let l:content = a:content
+        endif
+
+        " setbufline will append if a:num is the line after the last.
+        " nvim_buf_set_lines needs different arguments to do so.
+        if a:lnum == 1 + len(getbufline(a:buf_name, 1, '$'))
```
[the remark this answers is not in the document]
Comment (idanarye) [reply autoload/ctrlsf/buf.vim]: Wasn't sure if it's OK to do such a massive change here...

[the remark this answers is not in the document]
Comment (idanarye) [reply autoload/ctrlsf/buf.vim]: Actually - I can replace it with `nvim_buf_line_count`!

**Comment by dyng** [comment]
LGTM. Thanks for your contribution!

**Pull request merged**